src/Discovery/index.tsx

```diff
@@ -167,16 +200,22 @@ const DiscoveryWithMDSBackend: React.FC<{
 
     // indicate discovery tag is active even if we didn't click a button to get here
     props.onDiscoveryPageActive();
-  }, []);
+  }, [props, numberOfBatchesLoaded, numberOfStudiesForSmallerBatch]);
 
   let studyRegistrationValidationField = studyRegistrationConfig?.studyRegistrationValidationField;
   if (!isEnabled('studyRegistration')) {
     studyRegistrationValidationField = undefined;
   }
+
+  const allBatchesAreLoaded = studies === null
+    ? false
+    : (studies as Array<any>)?.length > totalNumberOfStudiesFromSmallBatches;
```

Comment:
better to check if `numberOfBatchesLoaded === expectedNumberOfTotalBatches`? since the current logic won't work if the total number of studies in MDS is less then `totalNumberOfStudiesFromSmallBatches` ?
although if that is the case we probably will disable this feature, but just to be on the safe side

Reply:
I don't believe this will work - the proposed approach sets the allBatchesAreLoaded variable to true before allowing the browser to process all the studies (such as with the function `populateStudiesWithConfigInfo`). This can take a couple seconds due to the number of studies. Setting this variable true before this has completed causes the progress bar to finish too soon.

However to improve clarity I have updated to the variable name to:
`allBatchesAreReady`